fuel-core/src/database.rs

```diff
@@ -297,10 +307,28 @@ impl Default for Database {
     }
 }
 
-impl BlockHeightDb for Database {
+impl BlockDb for Database {
     fn block_height(&self) -> anyhow::Result<BlockHeight> {
         Ok(self.get_block_height()?.unwrap_or_default())
     }
+
+    fn seal_block(
+        &mut self,
+        block_id: BlockId,
+        consensus: FuelBlockConsensus,
+    ) -> anyhow::Result<()> {
+        if self
+            .storage::<SealedBlockConsensus>()
+            .contains_key(&block_id.into())?
+        {
+            return Err(anyhow!("block {:x} is already sealed", &block_id))
+        }
+
+        self.storage::<SealedBlockConsensus>()
+            .insert(&block_id.into(), &consensus)
+            .map(|_| ())
+            .map_err(Into::into)
+    }
```

Comment:
This is a race condition because they are not in the same database transaction. Does it actually matter if the block is sealed twice?

Reply:
Hmm yeah it probably doesn't matter if its inserted twice. Will need to come up with a way to manage db transactions across service calls in the future though (maybe task local storage or similar).